src/pages/library/DataBrowserPreview.js

```diff
@@ -118,9 +118,24 @@ const DataBrowserPreview = ({ id }) => {
   Utils.useOnMount(() => {
     const loadData = async () => {
       const metadata = await Ajax(signal).DataRepo.getPreviewMetadata(id)
+
       setTables(metadata.tables)
-      selectTable({ value: metadata.tables[0].name })
+      setSelectOptions(_.flow(
```

Comment:
For ease of reading, would you consider pulling this flow out to a `const` that you can declare right under `metadata`?

Reply:
Sure, i actually had it that way initially, but then decided the team would rather the concise code! XD

Reply:
Im having an issue where I can't select the the first select option automatically, because the `setTables` method is async - how would you recommend solving that issue btw?